src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,266 @@
+    }
+    const previewContainer = $$(this.root).find('.coveo-preview-container');
+    if (previewContainer) {
+      this.updatePreviewContainer(previewContainer);
+    }
+  }
+
+  private buildPreviewHeader(suggestion: string) {
+    const text = `${this.options.headerText} "${suggestion}"`;
+    const header = $$('h4', { className: 'coveo-preview-header' }, text).el;
+    this.previewContainer.appendChild(header);
+  }
+
+  private buildResultsContainer() {
+    return $$('div', { className: 'coveo-preview-results' }).el;
+  }
+
+  private get previewContainer() {
+    return $$(this.root).find('.coveo-preview-container');
   }
 
   private querySuggestGetFocus(args: IQuerySuggestSelection) {
-    if (!this.shouldShowPreviewResults) {
+    if (!this.previewContainer) {
       return;
     }
-    this.executeQueryHover(args.suggestion);
+
+    if (this.previousSuggestionHovered === args.suggestion) {
+      return;
+    }
+
+    this.previousSuggestionHovered = args.suggestion;
+    this.timer && clearTimeout(this.timer);
+    this.timer = setTimeout(() => {
+      this.executeQueryHover(args.suggestion);
+    }, this.options.executeQueryDelay);
   }
 
-  private executeQueryHover(suggestion: string) {
+  private async executeQueryHover(suggestion: string) {
     const previousQueryOptions = this.queryController.getLastQuery();
     previousQueryOptions.q = suggestion;
     previousQueryOptions.numberOfResults = this.options.numberOfPreviewResults;
-    //TODO: I will need to execute a query, with the result,
-    //      build a container and display the result  next to the querySuggest
+    const results = await this.queryController.getEndpoint().search(previousQueryOptions);
+    $$(this.previewContainer).empty();
+    this.currentlyDisplayedResults = [];
+    if (!results) {
+      return;
+    }
+    this.buildPreviewHeader(suggestion);
+    this.buildResultsPreview(results);
+  }
+
+  private async buildResultsPreview(results: IQueryResults) {
+    const builtResults = this.buildResultsContainer();
+    this.previewContainer.appendChild(builtResults);
+    this.setupRenderer(builtResults);
+    const HTMLResult = await this.templateToHtml.buildResults(results, 'preview', this.currentlyDisplayedResults);
+    if (!(HTMLResult.length > 0)) {
+      return;
+    }
+    this.updateResultPerRow(HTMLResult);
+    this.renderer.renderResults(HTMLResult, true, result => {});
+  }
+
+  private updateResultPerRow(elements: HTMLElement[]) {
+    let resultAvailableSpace = '33%';
+    if (elements.length === 4) {
+      resultAvailableSpace = '50%';
+    }
+    elements.forEach(element => {
+      this.updateFlexCSS(element, resultAvailableSpace);
+    });
+  }
+
+  private updateFlexCSS(element: HTMLElement, value: string) {
+    element.style.flex = `0 0 ${value}`;
+  }
+
+  private setupRenderer(resultsContainer: HTMLElement) {
+    const rendererOption: IResultListOptions = {
+      resultContainer: resultsContainer
```

Comment:
should be both `resultsContainer` and you could just declare it once
```
const rendererOption: IResultListOptions = {
      resultsContainer
...

Reply:
ok I'll rename the value in the interface